Enc28j60 fixes

tbursztyka opened the pull request:
Please review

Commit:
samples/telnet: Fix accessing iface ipv4 config

A left over from commit 47dafffb67d72d75610401ae76c3be69ed962fab

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
drivers/ethernet: Fix and clean a bit ENC28J60 driver

Driver is still a bit messy, not using for instance the 0-copy
capabilities of SPI API, but this will be fix later.

Constify most of the spi buf structures, removing useless variables,
renaming function with common prefix eth_enc28j60_ etc...

It seems to fix an issue on verifying if tx was successful as well.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

jukkar reviewed (commented):
Could you split the patch, the telnet fix could be applied already?

tbursztyka commented:
I'll come with the 0-copy fix today, so no worry: all will be applied today.

Commit:
drivers/ethernet: Optimize memory read/write operations in ENC28J60

SPI API helps to directly transfer bytes from/to relevant buffers, so
let's take advantage of it.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
drivers/ethernet: Reduce runtime context size in ENC28J60 driver

- tx_tsv is never used anywhere
- and rx_rsv can be allocated on stack

Optimizing a bit the stack usage in eth_enc28j60_rx() function as well.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>
